### ISSUE TITLE
Template should support more than one source type

### ISSUE THREAD

**Issue by AitorAlgorta** [airyhq/airy#1206]
**Should templates support more than one source type?**

Since some sources share payloads and template types it would be nice to just have one template that can be sent to all the sources that support that kind of payload. For example RichCards in Google and ChatPlugin.

**Comment by lucapette** [comment]
I think the answer should be "No they shouldn't"

It would save some very little space but complicate the api surface (as we'd have to take that into account in more than one endpoint). I feel like this is a case of "a little duplication is better than the wrong abstraction". Maybe "wrong" strong work for this context...)

**Comment by AitorAlgorta** [comment]
but what about text templates, they work for any `sourceType` and they are used a lot. We cannot have a text template for each source, it will look super awkward.

**Comment by lucapette** [comment]
Technically that's not true right? I don't think there's _really_ a template that works for any source type. The payloads of each source differ even in the simplest cases (which is maybe good for us because it's a decent design constraint).

Even if we'd have such a case, I'd still argue against it though. As we do not even "mix" templates from different sources. The information is always presented (in the UI at least) in a source type specific context. 

But it's good we're having this conversation as I believe that means the `/templates.list` is not "correct" at the moment as it should require the type and it doesnt (I assumed we fixed this when we figured the same problem with create) /cc @paulodiniz
